teuthology/openstack/__init__.py

```diff
@@ -617,6 +617,59 @@ def main(self):
                 self.teardown()
         return exit_code
 
+    def upload_yaml_file(self, fp):
+        """
+        Given an absolute path fp, assume it is a YAML file existing
+        on the local machine and upload it to the remote teuthology machine
+        (see https://github.com/SUSE/teuthology/issues/56 for details)
+        """
+        f = open(fp, 'r') # will throw exception on failure
+        f.close()
+        log.info("Detected local YAML file {}".format(fp))
+        machine = self.username + "@" + self.instance.get_floating_ip_or_ip()
+
+        def ssh_command(s):
+            return "ssh -i {k} {m} sh -c \\\"{s}\\\"".format(
+                k=self.key_filename,
+                m=machine,
+                s=s,
+            )
+
+        log.info("Uploading local file {} to teuthology machine".format(fp))
+        remote_fp=os.path.normpath(
+            '/home/{un}/yaml/{fp}'.format(
+                un=self.username,
+                fp=fp,
+            )
+        )
+        command = ssh_command("stat {aug_fp}".format(
+            aug_fp=remote_fp,
+        ))
+        try:
+            misc.sh(command)
```

Comment:
I looked at `TeuthologyOpenStack.ssh()` and have no idea how it works. I prefer `misc.py` because it logs the command and throws an exception if the command doesn't succeed.